btf/types_test.go

```diff
@@ -0,0 +1,73 @@
+package btf
```

Comment:
Are there tests for circular types, nested types that would exceed the 32 limit, or other corner / special cases that we handle specially?

Reply:
No, this is woefully under tested unfortunately. I'll try to add some that cover the trickiest bits (like the circular rules).

Reply:
I half punted on this by not doing much validation at all. Cyclical types are in fact required, and I don't understand what constraints are required, yet. For now, all callers will have to do their own validation. I think that's ok since the package is hidden for now.